pkg/profile/helm.go

```diff
@@ -0,0 +1,111 @@
+package profile
+
+import (
+	"reflect"
+	"strings"
+
+	helmv2 "github.com/fluxcd/helm-controller/api/v2beta1"
+	sourcev1 "github.com/fluxcd/source-controller/api/v1beta1"
+	profilesv1 "github.com/weaveworks/profiles/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HelmReleaseRequiresUpdate checks if the helm release resource requires updating
+func HelmReleaseRequiresUpdate(existingRes, newRes *helmv2.HelmRelease) bool {
+	switch {
+	case existingRes.Spec.Chart.Spec.Chart != newRes.Spec.Chart.Spec.Chart:
+		return true
+	case existingRes.Spec.Chart.Spec.Version != newRes.Spec.Chart.Spec.Version:
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.Chart.Spec.SourceRef, newRes.Spec.Chart.Spec.SourceRef):
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.Values, newRes.Spec.Values):
+		return true
+	case !reflect.DeepEqual(existingRes.Spec.ValuesFrom, newRes.Spec.ValuesFrom):
+		return true
+	default:
+		return false
+	}
+}
+
+// HelmRepoRequiresUpdate checks if the helm repository resource requires updating
+func HelmRepoRequiresUpdate(existingRes, newRes *sourcev1.HelmRepository) bool {
+	switch {
+	case existingRes.Spec.URL != newRes.Spec.URL:
+		return true
+	default:
+		return false
+	}
```

Comment:
```suggestion
	return existingRes.Spec.URL != newRes.Spec.URL:
```

Reply:
This is fair, I'd be inclined to stick with the select since thats the pattern used across the update checkers, but I could be convinced here... :shrug: